sigopt/xgboost/experiment.py

```diff
@@ -36,17 +47,52 @@ def parse_and_create_experiment(self):
         else:
           metric['name'] = DEFAULT_EVALS_NAME + '-' + metric['name']
 
+  def parse_and_create_parameters(self):
+    # Set defaults as needed
+    if 'budget' not in self.experiment_config_parsed:
+      self.experiment_config_parsed['budget'] = DEFAULT_BO_ITERATIONS
+    if 'parallel_bandwidth' not in self.experiment_config_parsed:
+      self.experiment_config_parsed['parallel_bandwidth'] = 1
+    if 'type' not in self.experiment_config_parsed:
+      self.experiment_config_parsed['type'] = 'offline'
+
+    # Parse params
+    if 'parameters' not in self.experiment_config_parsed:
+      self.experiment_config_parsed['parameters'] = [
+        SEARCH_BOUNDS[SEARCH_PARAMS.index(param_name)] for param_name in DEFAULT_SEARCH_PARAMS
+      ]
+    else:
+      for param in self.experiment_config_parsed['parameters']:
+        if 'bounds' not in param:
+          param_name = param['name']
+          assert param_name in SEARCH_PARAMS, f'We do not support autoselection of bounds for {param_name}'
+          search_bound = SEARCH_BOUNDS[SEARCH_PARAMS.index(param_name)]
+          param['bounds'] = search_bound['bounds']
+          param['type'] = search_bound['type']
+          if 'transformation' in search_bound:
+            param['transformation'] = search_bound['transformation']
+
+  def parse_and_create_experiment_config(self):
+    self.experiment_config_parsed = copy.deepcopy(self.experiment_config)
+    self.parse_and_create_metrics()
+    self.parse_and_create_parameters()
+
+  def parse_and_create_sigopt_experiment(self):
     # Check key overlap between parameters to be optimized and parameters that are set
-    params_optimized = [param['name'] for param in self.experiment_config['parameters']]
-    assert len(set(params_optimized) & set(self.params.keys())) == 0, (
-      'There is overlap between optimized params and user-set params'  #TODO: more descriptive error message
+    params_optimized = [param['name'] for param in self.experiment_config_parsed['parameters']]
+    params_overlap = list(set(params_optimized) & set(self.params.keys()))
+    assert len(params_overlap) == 0, (
+      f'There is overlap between tuned parameters and user-set parameters: {params_overlap}. '
+      f'Parameter names cannot be defined in both locations'
     )
 
     # Check that num_boost_round is not set by both sigopt experiment and user
     if self.num_boost_round:
-      assert 'num_boost_round' not in params_optimized
+      assert 'num_boost_round' not in params_optimized, \
+        'We are optimizing num_boost_round, and right now it is fixed. Please remove it from either the search space' \
+        ' or the input arguments. '
```

Comment:
Could also just move 82-93 under `self.parse_and_create_parameters()` and get rid of this function.